### ISSUE TITLE
CPC and Detail Enhancer disable?

### ISSUE THREAD

**Issue by vmovups** [Maassoft/ColorControl#399]
LG has implemented a processing named "CPC" that causes a vignetting effect(pixels away from center are darker, with banding) visible on large areas of gray, i can trigger it on all modes except FILMMAKER and ISF Bright for some reason.
EZADJUST→Option5/Key→CPC is rumored to be the option that controls it.
On my G4 it can be changed to the following values : 35(factory default), 30, 25, Default(confusingly named, doesn't appear to disable the vignetting)
While searching for what it does i've found that people with LG WOLED monitors were able to completely turn off the vignetting with the service remote and changing CPC, i wonder if there is any way ColorControl can be made to turn it off too?(or at least change it like TPC, GSR)

There is also Detail Enhancer, LG introduced this in all 2024 models and markets it as a revolutionary anti-banding features but it's just PWM checkerboard dithering to reduce overshoot and it degrades the picture quality in dark parts of the image even if static. 
Is there any way to find if it can be toggled off similarly to how the service menu flag is hidden but can be toggled remotely through ColorControl?
Also thank you for the program, being able to change TPC/GSR without having to fetch the service remote is super convenient, and now that it's impossible without ColorControl since LG added a service menu flag it is a lifesaver for usecases where TPC/GSR are undesirable.

**Comment by vmovups** [comment]
Adding attachments to better illustrate Detail Enhancer : 
![LG G4 with Detail Enhancer](https://github.com/user-attachments/assets/2e884d70-88f9-4601-b754-4ca1d468544a)(subpixels are turned off and on in a pattern, image isn't temporally stable and diagonal lines are visible in certain conditions)
![Older model without Detail Enhancer](https://github.com/user-attachments/assets/18c18721-04e2-4661-8836-3562e2580f8f)(normal pixel grid, image is temporally stable, no artifact visible)
I can upload images of CPC in action if desired.

**Comment by error812** [comment]
Hi. so have you found any solution to disable detail enhancer?

**Comment by vmovups** [comment]
> Hi. so have you found any solution to disable detail enhancer?

Sadly no, Samsung QD-OLED monitors have a similar dithering algorithm but much better quality(the threshold is much lower compared to LG, will not trigger on bright whites and uses a vertical line pattern of brighter and dimmer pixels)

It seems embedded in their "processor", i know because C1 with 2024 replacement panel has the C4's magenta and green whites but no detail enhancer or CPC.

It's sad that the reviewers don't care, and people won't say anything, the picture quality is being massively downgraded and they're asking for more, to our dismay.

You can still reduce CPC by setting it to Default in the real service menu, else you can use FILMMAKER mode but it seems it has more latency than game mode?

I ended up getting a refund, no way i was buying a TV that's brighter but still ends up worse than one with a WBC panel.

**Comment by error812** [comment]
It is sad that as you've said, not one single reviewer has acknowledged this issue, which is baffling for me. I don't have the best eyes in the world, but the diethering artifacts were seen in the first hour of gaming. So either all reviewers are payed by LG to say the sweetest things about their product, or they are allmost blind.

I decided to keep mine, just because I got it heavily discounted and the S95D has other issues that I don't want to confront like connection issues, antiglare coating etc.

I hope that at some point, common sense rational people working at LG (if there are any) will release a fix, which would be to disable atleast in game mode,this non sense horrid processing effect. But since it hasn't happened by now, I guess there are very slim chances that it will ever happen.

**Comment by vmovups** [comment]
I don't believe LG will do anything until they find a way to improve the OLED matrix itself, the cause i believe is because of the resulting overshoot and banding artifacts in dark content on the G3 when they improved efficiency and brightness without improving the matrix's ability to produce low voltages accurately and without overshoot.
However CPC has zero excuse to exist as is without being able to turn it off as we wish, it's a power saving mechanism. Ideally LG would have let people change it if they didn't like it but they didn't...
